synthesizer/src/snark/proving_key/mod.rs

```diff
@@ -68,7 +64,13 @@ impl<N: Network> ProvingKey<N> {
         let timer = std::time::Instant::now();
 
         // Compute the batch proof.
-        let batch_proof = Proof::new(Marlin::<N>::prove_batch(N::marlin_fs_parameters(), self, assignments, rng)?);
+        let mut assignment_refs = vec![];
+        for assignment in assignments {
+            assignment_refs.push(assignment);
+        }
```

Comment:
why do this instead of just passing `assignments` in L72?

Reply:
Because prove_batch expects `&[&circuit::Assignment<>]` not `&[circuit::Assignment<>]`